Protocol conformances in WebKitSwift needlessly use @retroactive

aestes opened the pull request:
#### 55b925eb3c6b6d494733e97cc3095bd518f993b9
<pre>
Protocol conformances in WebKitSwift needlessly use @retroactive
<a href="https://bugs.webkit.org/show_bug.cgi?id=294955">https://bugs.webkit.org/show_bug.cgi?id=294955</a>
<a href="https://rdar.apple.com/154261967">rdar://154261967</a>

Reviewed by Richard Robinson.

WebKitSwift vends Objective-C interfaces declared in headers imported by the WebKitSwift.h umbrella
header, but implements these interfaces in Swift using @objc @implementation extensions. In order
for the Swift compiler to see the Objective-C declarations, WebKitSwift defines a clang module
named WebKitSwift that imports WebKitSwift.h, and Swift files containing the implementations import
the WebKitSwift clang module. Since the WebKitSwift Swift module is named libWebKitSwift,
technically the Objective-C declarations come from a different module than their implementations.

This presents a problem when we attempt to conform these types to a protocol imported from another
module. Since the Swift compiler thinks both the interface and the protocol are from imported
modules, a &quot;extension declares a conformance of imported type &apos;Foo&apos; to imported protocol &apos;Bar&apos;&quot;
warning is emitted. Previously we have silenced this warning by annotating the conformance with
@retroactive, but it turns out this is unnecessary.

Similar to how Swift overlays are implemented in mixed-source frameworks, we can convince Swift
that our Objective-C interfaces come from the same module that implements them by giving both the
clang and Swift modules the same name (WebKitSwift), and importing the clang module via
`@_exported import WebKitSwift`. This allows us to remove the needless @retroactive attributes.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift:
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaPlayer.swift:
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift:
* Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift:
* Source/WebKit/WebKitSwift/Preview/WKPreviewWindowController.swift:
* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift:
* Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
* Source/WebKit/WebKitSwift/TextAnimation/WKTextAnimationManagerIOS.swift:
* Source/WebKit/WebKitSwift/WebKitSwift.swift: Copied from Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift.
* Source/WebKit/WebKitSwift/WritingTools/IntelligenceTextEffectViewManager.swift:
* Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceSmartReplyTextEffectCoordinator.swift:

Canonical link: <a href="https://commits.webkit.org/296629@main">https://commits.webkit.org/296629@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9e255a029f586b1335ccc55c1eb0f6ce0738a724

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109105 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28765 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19190 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114316 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59404 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111068 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29448 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37332 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82916 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112053 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23425 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98267 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63361 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22826 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16409 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/58998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92796 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16452 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117434 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36153 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26724 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91931 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36524 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94531 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91737 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23357 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36649 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14401 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32011 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36050 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41555 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35744 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39082 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37429 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift:
```diff
@@ -23,6 +23,8 @@
 
 #if ENABLE_MEDIA_SESSION_COORDINATOR && HAVE_GROUP_ACTIVITIES
 
+@_exported import WebKitSwift
```
is this actually needed in every file? _WebKit_SwiftUI does a similar thing, but I only have it in a single dedicated file there

aestes committed:
Protocol conformances in WebKitSwift needlessly use @retroactive
https://bugs.webkit.org/show_bug.cgi?id=294955
rdar://154261967

Reviewed by Richard Robinson.

WebKitSwift vends Objective-C interfaces declared in headers imported by the WebKitSwift.h umbrella
header, but implements these interfaces in Swift using @objc @implementation extensions. In order
for the Swift compiler to see the Objective-C declarations, WebKitSwift defines a clang module
named WebKitSwift that imports WebKitSwift.h, and Swift files containing the implementations import
the WebKitSwift clang module. Since the WebKitSwift Swift module is named libWebKitSwift,
technically the Objective-C declarations come from a different module than their implementations.

This presents a problem when we attempt to conform these types to a protocol imported from another
module. Since the Swift compiler thinks both the interface and the protocol are from imported
modules, a "extension declares a conformance of imported type 'Foo' to imported protocol 'Bar'"
warning is emitted. Previously we have silenced this warning by annotating the conformance with
@retroactive, but it turns out this is unnecessary.

Similar to how Swift overlays are implemented in mixed-source frameworks, we can convince Swift
that our Objective-C interfaces come from the same module that implements them by giving both the
clang and Swift modules the same name (WebKitSwift), and importing the clang module via
`@_exported import WebKitSwift`. This allows us to remove the needless @retroactive attributes.

* Source/WebKit/Configurations/WebKitSwift.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebKitSwift/GroupActivities/WKGroupSession.swift:
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaPlayer.swift:
* Source/WebKit/WebKitSwift/LinearMediaKit/LinearMediaTypes.swift:
* Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift:
* Source/WebKit/WebKitSwift/Preview/WKPreviewWindowController.swift:
* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift:
* Source/WebKit/WebKitSwift/StageMode/WKStageMode.swift:
* Source/WebKit/WebKitSwift/TextAnimation/WKTextAnimationManagerIOS.swift:
* Source/WebKit/WebKitSwift/WebKitSwift.swift: Copied from Source/WebKit/WebKitSwift/MarketplaceKit/WKMarketplaceKit.swift.
* Source/WebKit/WebKitSwift/WritingTools/IntelligenceTextEffectViewManager.swift:
* Source/WebKit/WebKitSwift/WritingTools/PlatformIntelligenceTextEffectView.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceReplacementTextEffectCoordinator.swift:
* Source/WebKit/WebKitSwift/WritingTools/WKIntelligenceSmartReplyTextEffectCoordinator.swift:

Canonical link: https://commits.webkit.org/296629@main
